### PR TITLE
Problem: getlog REST API - PQSWMBT-3457

### PR DESCRIPTION
**Pull request by jeffdegott**
Solution: reworked to support tar arch. and journalctl logfile(s) generation
* new emc4j-audit, journald and ALL basenames, code reworked to support generated or archive logfile
Signed-off-by: Degott, Francois Regis <FrancoisRegisDegott@eaton.com>

### CODE REVIEW

File: src/web/src/getlog_GET.ecpp [hunk 1]
```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
...
+          // This is in newer codebase : _deflator.reinitialize(); which does "deflateReset(&_stream)"
           _crc = 0;
           _size = 0;
         }
 }; // class _clonedcode_Compressor
 
+// journald logs export to logfile
+int exec_journalctl_command (const std::string& logfile)
+{
+    // 'www-data' user must be owned by 'systemd-journal' group,
```
Comment (aquette) [inline src/web/src/getlog_GET.ecpp]: 'www-data' user must be *part of the* 'systemd-journal' group,

File: src/web/src/getlog_GET.ecpp [hunk 2]
```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
...
+    // to let journalctl access journald files (see fty-core)
+    // usermod -a -G systemd-journal www-data
+    // avoid wrapper usage /usr/libexec/bios/journalctl
```
Comment (aquette) [inline src/web/src/getlog_GET.ecpp]: Also avoid using 42ITy wrapper (/usr/libexec/bios/journalctl) by calling the real version from /bin

File: src/web/src/getlog_GET.ecpp [hunk 3]
```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
...
+    // supported (baseName, filePath) pairs
+    struct {
+        std::string baseName;
+        std::string filePath;
+    } logsData [] = {
+        { "messages",       "/var/log/messages" },
+        { "www-audit",      "/var/log/tntnet/www-audit.log" },
+        { "emc4j-audit",    "/var/log/etn-emc4j-ipm2/karaf.log" },
+        { "journald",       "/tmp/journald.log" }, // gen. by journalctl
+        { "ALL",            "/tmp/getlog_all.tar" }, // archive,all logfiles above
```
Comment (aquette) [inline src/web/src/getlog_GET.ecpp]: any reason to use upper case for "ALL" and not the others?

File: src/web/src/getlog_GET.ecpp [hunk 4]
```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
...
 
-        if (logname_ext == "" || logname_ext == ".txt") {
-            std::ifstream in( logfile.c_str() );
-            if (!in)
-                throw std::runtime_error(TRANSLATE_ME ("Could not open requested logfile: ") + logfile);
+        if (logname_base == "ALL") {
+            // make a tar archive (logfile) owning *all* logfiles
+            delete_logfile = true;
+
+            std::string tmpdir = "/tmp/IPM2-logs";
```
Comment (aquette) [inline src/web/src/getlog_GET.ecpp]: so playing in the /tmp world in the end?!

File: src/web/src/getlog_GET.ecpp (hunk 3 above)
Comment (jeffdegott) [reply src/web/src/getlog_GET.ecpp]: not really, just a manner to strongly differentiate the 'all logs' feature.

File: src/web/src/getlog_GET.ecpp (hunk 4 above)
Comment (jeffdegott) [reply src/web/src/getlog_GET.ecpp]: yep

**Comment by jimklimov** [comment]
Big Please: next time separate the usually big style changes (like comment chars... why?!) from real code changes. Might be same PR, different commits. That would make reviews a lot easier, looking at a few dozen lines instead of 400 or so.

**Comment by jeffdegott** [comment]
> 
> 
> Big Please: next time separate the usually big style changes (like comment chars... why?!) from real code changes. Might be same PR, different commits. That would make reviews a lot easier, looking at a few dozen lines instead of 400 or so.

Sorry for that Jim

**Pull request merged**